<!DOCTYPE html PUBLIC "-//W3C//DTD XHTML 1.0 Transitional//EN" "https://www.w3.org/TR/xhtml1/DTD/xhtml1-transitional.dtd">
<html xmlns="http://www.w3.org/1999/xhtml">
<head>
<meta http-equiv="Content-Type" content="text/xhtml;charset=UTF-8"/>
<meta http-equiv="X-UA-Compatible" content="IE=11"/>
<meta name="generator" content="Doxygen 1.9.3"/>
<meta name="viewport" content="width=device-width, initial-scale=1"/>
<title>ntHash: README</title>
<link href="tabs.css" rel="stylesheet" type="text/css"/>
<script type="text/javascript" src="jquery.js"></script>
<script type="text/javascript" src="dynsections.js"></script>
<link href="search/search.css" rel="stylesheet" type="text/css"/>
<script type="text/javascript" src="search/searchdata.js"></script>
<script type="text/javascript" src="search/search.js"></script>
<link href="doxygen.css" rel="stylesheet" type="text/css" />
</head>
<body>
<div id="top"><!-- do not remove this div, it is closed by doxygen! -->
<div id="titlearea">
<table cellspacing="0" cellpadding="0">
 <tbody>
 <tr id="projectrow">
  <td id="projectalign">
   <div id="projectname">ntHash
   </div>
  </td>
 </tr>
 </tbody>
</table>
</div>
<!-- end header part -->
<!-- Generated by Doxygen 1.9.3 -->
<script type="text/javascript">
/* @license magnet:?xt=urn:btih:d3d9a9a6595521f9666a5e94cc830dab83b65699&amp;dn=expat.txt MIT */
var searchBox = new SearchBox("searchBox", "search",'Search','.html');
/* @license-end */
</script>
<script type="text/javascript" src="menudata.js"></script>
<script type="text/javascript" src="menu.js"></script>
<script type="text/javascript">
/* @license magnet:?xt=urn:btih:d3d9a9a6595521f9666a5e94cc830dab83b65699&amp;dn=expat.txt MIT */
$(function() {
  initMenu('',true,false,'search.php','Search');
  $(document).ready(function() { init_search(); });
});
/* @license-end */
</script>
<div id="main-nav"></div>
<!-- window showing the filter options -->
<div id="MSearchSelectWindow"
     onmouseover="return searchBox.OnSearchSelectShow()"
     onmouseout="return searchBox.OnSearchSelectHide()"
     onkeydown="return searchBox.OnSearchSelectKey(event)">
</div>

<!-- iframe showing the search results (closed by default) -->
<div id="MSearchResultsWindow">
<iframe src="javascript:void(0)" frameborder="0" 
        name="MSearchResults" id="MSearchResults">
</iframe>
</div>

</div><!-- top -->
<div><div class="header">
  <div class="headertitle"><div class="title">README </div></div>
</div><!--header-->
<div class="contents">
<div class="textblock"><p ><a href="https://github.com/bcgsc/ntHash/releases"><img src="https://img.shields.io/github/release/bcgsc/ntHash.svg" alt="Release" style="pointer-events: none;" class="inline"/></a> <a href="https://github.com/bcgsc/ntHash/archive/master.zip"><img src="https://img.shields.io/github/downloads/bcgsc/ntHash/total?logo=github" alt="Downloads" class="inline"/></a> <a href="https://github.com/bcgsc/ntHash/issues"><img src="https://img.shields.io/github/issues/bcgsc/ntHash.svg" alt="Issues" style="pointer-events: none;" class="inline"/></a></p>
<p ><img src="nthash-logo.png" alt="Logo" class="inline"/></p>
<p >ntHash = ntHash is a recursive hash function for hashing all possible k-mers in a DNA/RNA sequence.</p>
<h1><a class="anchor" id="autotoc_md0"></a>
Installation &amp; Usage</h1>
<h2><a class="anchor" id="autotoc_md1"></a>
btllib (Recommended)</h2>
<p >ntHash is included in <a href="https://github.com/bcgsc/btllib">btllib</a>, which can be installed using Conda:</p>
<div class="fragment"><div class="line">conda install -c bioconda btllib</div>
</div><!-- fragment --><p >This installs other classes such as Bloom filters that can be used with ntHash. Also, ntHash can be imported in Python projects via btllib's wrappers. ntHash's code is automatically synced to btllib on each release. Refer to <a href="https://github.com/bcgsc/btllib/blob/master/README.md">btllib's readme</a> for usage instructions.</p>
<h2><a class="anchor" id="autotoc_md2"></a>
Manual Compilation</h2>
<p >To use ntHash without installing btllib, download the repo and generate a cmake buildsystem in an arbitrary directory (e.g. <code>release</code>), by running the following command in the project's root:</p>
<div class="fragment"><div class="line">cmake -S . -B release</div>
</div><!-- fragment --><p >Then, build the project and its dependencies using:</p>
<div class="fragment"><div class="line">cmake --build release --target all</div>
</div><!-- fragment --><p >The project's executable (<code>./nthash</code>) and static library (<code>libnthash.a</code>) will be generated in the <code>release</code> folder.</p>
<h3><a class="anchor" id="autotoc_md3"></a>
Executable</h3>
<p >After building the project, use the output binary file to generate hash values from input data and store the results on disk:</p>
<div class="fragment"><div class="line">Usage: nthash [options] files </div>
<div class="line"> </div>
<div class="line">Positional arguments:</div>
<div class="line">files           Input sequence files [required]</div>
<div class="line"> </div>
<div class="line">Optional arguments:</div>
<div class="line">-v --version    prints version information and exits [default: false]</div>
<div class="line">-k              k-mer size [required]</div>
<div class="line">-o              Output file (for -f collect) or directory path [required]</div>
<div class="line">-f              Output file organization (create files containing hashes for each &#39;file&#39;, &#39;record&#39;, or &#39;collect&#39; all hashes into a single file [file] [default: &quot;file&quot;]</div>
<div class="line">-h              Number of hashes per k-mer [1] [default: 1]</div>
<div class="line">-s              Input spaced seed patterns separated by commas (e.g. 1110111,11011011). Performs k-mer hashing if no value provided.</div>
<div class="line">--long          Optimize file reader for long sequences (&gt;5kbp) [default: false]</div>
<div class="line">--binary        Output hashes in binary files (otherwise plain text) [default: false]</div>
<div class="line">--verbose       Print progress to stdout [default: false]</div>
</div><!-- fragment --><p >For example, given two input files <code>1.fa</code> and <code>2.fa</code>, two hash values for each 64-mer can be saved to a binary file called <code>out.bin</code> by running this command in <code>release</code>:</p>
<div class="fragment"><div class="line">./nthash -k 64 -h 2 -o out.bin -f collect --binary --verbose 1.fa 2.fa</div>
</div><!-- fragment --><h3><a class="anchor" id="autotoc_md4"></a>
Static Library Usage</h3>
<p >To use ntHash in a C++ project:</p><ul>
<li>Link the code with <code>nthashlib.a</code> (i.e. pass <code>-L path/to/nthash/release -l nthashlib.a</code> to the compiler).</li>
<li>Add the <code>include</code> directory (pass <code>-I path/to/nthash/include</code> to the compiler).</li>
<li>Import ntHash in the code using <code>#include &lt;nthash/nthash.hpp&gt;</code>.</li>
<li>Access ntHash classes from the <code>nthash</code> namespace.</li>
</ul>
<p >Refer to docs for more information.</p>
<h1><a class="anchor" id="autotoc_md5"></a>
Publications</h1>
<h2><a class="anchor" id="autotoc_md6"></a>
&lt;a href="http://bioinformatics.oxfordjournals.org/content/early/2016/08/01/bioinformatics.btw397" &gt;ntHash&lt;/a&gt;</h2>
<p >Hamid Mohamadi, Justin Chu, Benjamin P Vandervalk, and Inanc Birol. <b>ntHash: recursive nucleotide hashing</b>. <em>Bioinformatics</em> (2016) 32 (22): 3492-3494. <a href="http://dx.doi.org/10.1093/bioinformatics/btw397">doi:10.1093/bioinformatics/btw397</a></p>
<h1><a class="anchor" id="autotoc_md7"></a>
Acknowledgements</h1>
<p >This projects uses:</p><ul>
<li><a href="https://github.com/philsquared/Catch">CATCH</a> unit test framework for C/C++ </li>
</ul>
</div></div><!-- contents -->
</div><!-- PageDoc -->
<!-- start footer part -->
<hr class="footer"/><address class="footer"><small>
Generated by&#160;<a href="https://www.doxygen.org/index.html"><img class="footer" src="doxygen.svg" width="104" height="31" alt="doxygen"/></a> 1.9.3
</small></address>
</body>
</html>
